### PR TITLE
Add an OBO controller to the web api dev app

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs [hunk 1]
```diff
@@ -80,7 +85,11 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             // No AT in the cache or AT needs to be refreshed
             try
             {
-                return await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                using (logger.LogBlockDuration("[OBO request] Fetching OBO token from ESTS"))
+                {
+                    var result = await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs:90]: Can you simply return without using the result?

File: tests/devapps/WebApi/Controllers/OboController.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                s_traceSource.TraceEvent(TraceEventType.Error, 1, "##### FOUND!! " + requestId);
+
+
+                System.IO.File.WriteAllText($"c:\\temp\\obo_{requestId}.txt", sb.ToString());
+                System.IO.File.WriteAllText($"c:\\temp\\obo2_{requestId}.txt", sb2.ToString());
+
+            }
+
+
+            return res.AuthenticationResultMetadata.DurationTotalInMs;
```
Comment (trwalke) [inline tests/devapps/WebApi/Controllers/OboController.cs:94]: nit: There is a bunch of extra spacing around these lines above

**Review by trwalke — APPROVED** [review]
2 minor comments, otherwise LGTM

File: src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs]: I'm going to leave it in. it helps when debugging. the compiler will optimize it anyway, there's not perf impact.

**Review by neha-bhargava — APPROVED** [review]
LGTM

**Pull request merged**